From 79c8d6a2755c9dfa00a5e86378e89a94eef0504d Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 9 Aug 2022 23:57:03 -0700
Subject: [PATCH] Replace __attribute_malloc__ with 
 __attribute__((__malloc__))

__attribute_malloc__ is not available on musl

Fixes
| ../../git/auparse/auparse.h:54:2: error: expected function body after function declarator
|         __attribute_malloc__ __attr_dealloc (auparse_destroy, 1);
|         ^

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Upstream: https://github.com/linux-audit/audit-userspace/pull/311

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---

 auparse/auparse.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/auparse/auparse.h b/auparse/auparse.h
index 95cf256d..c7dbe5ff 100644
--- a/auparse/auparse.h
+++ b/auparse/auparse.h
@@ -51,7 +51,7 @@ typedef void (*auparse_callback_ptr)(auparse_state_t *au,
 void auparse_destroy(auparse_state_t *au);
 void auparse_destroy_ext(auparse_state_t *au, auparse_destroy_what_t what);
 auparse_state_t *auparse_init(ausource_t source, const void *b)
-	__attribute_malloc__ __attr_dealloc (auparse_destroy, 1);
+	__attribute__((__malloc__)) __attr_dealloc (auparse_destroy, 1);
 int auparse_new_buffer(auparse_state_t *au, const char *data, size_t data_len)
 	__attr_access ((__read_only__, 2, 3));
 int auparse_feed(auparse_state_t *au, const char *data, size_t data_len)
-- 
2.37.1

